### PR TITLE
Deprecate cross-reference in MERGE clause

### PR DESCRIPTION
**Pull request by WilcoNeo**
[Trello card](https://trello.com/c/gcaWiON3/2479-deprecate-cross-references-in-merge)

### CODE REVIEW

**Review by blouerat — APPROVED** [review]

**Review by JPryce-Aklundh — APPROVED** [review]
Thanks @WilcoNeo

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-cypher-950.surge.sh

Updated pages:
- [MERGE](https://neo4j-docs-cypher-950.surge.sh/cypher-manual/5/clauses/merge)
- [Deprecations, additions, and compatibility](https://neo4j-docs-cypher-950.surge.sh/cypher-manual/5/deprecations-additions-removals-compatibility)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-cypher-950.surge.sh

Updated pages:
- [MERGE](https://neo4j-docs-cypher-950.surge.sh/cypher-manual/5/clauses/merge)
- [Deprecations, additions, and compatibility](https://neo4j-docs-cypher-950.surge.sh/cypher-manual/5/deprecations-additions-removals-compatibility)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-cypher-950.surge.sh

Updated pages:
- [MERGE](https://neo4j-docs-cypher-950.surge.sh/cypher-manual/5/clauses/merge)
- [Deprecations, additions, and compatibility](https://neo4j-docs-cypher-950.surge.sh/cypher-manual/5/deprecations-additions-removals-compatibility)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-cypher-950.surge.sh

Updated pages:
- [MERGE](https://neo4j-docs-cypher-950.surge.sh/cypher-manual/5/clauses/merge)
- [Deprecations, additions, and compatibility](https://neo4j-docs-cypher-950.surge.sh/cypher-manual/5/deprecations-additions-removals-compatibility)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-cypher-950.surge.sh

Updated pages:
- [MERGE](https://neo4j-docs-cypher-950.surge.sh/cypher-manual/5/clauses/merge)
- [Deprecations, additions, and compatibility](https://neo4j-docs-cypher-950.surge.sh/cypher-manual/5/deprecations-additions-removals-compatibility)

**Comment by WilcoNeo** [comment]
This PR is ready for a new review :)

**Review by blouerat — APPROVED** [review]

**Review by JPryce-Aklundh — APPROVED** [review]

File: modules/ROOT/pages/clauses/merge.adoc [hunk 1]
```diff
@@ -373,6 +373,26 @@ Cannot merge the following relationship because of null property value for 'sinc
 ----
 ====
 
+[NOTE]
+====
+Specifying a property of an entity (node or relationship) by referring to the property of another entity in the same `MERGE` clause is deprecated.
```
Comment (JPryce-Aklundh) [inline modules/ROOT/pages/clauses/merge.adoc]:
```suggestion
As of Neo4j 5.20, specifying a property of an entity (node or relationship) by referring to the property of another entity in the same `MERGE` clause is deprecated.
```

File: modules/ROOT/pages/deprecations-additions-removals-compatibility.adoc [hunk 2]
```diff
@@ -19,6 +19,29 @@ Replacement syntax for deprecated and removed features are also indicated.
...
+label:functionality[]
+label:deprecated[]
+[source, cypher, role="noheader"]
+----
+MERGE (a {foo:1})-[:T]->(b {foo:a.foo})
+----
+a|
+
+Merging a node or relationship entity,
+and then referencing that entity in a property definition in the same `MERGE` clause is deprecated.
```
Comment (JPryce-Aklundh) [inline modules/ROOT/pages/deprecations-additions-removals-compatibility.adoc]:
```suggestion
Merging a node or relationship entity, and then referencing that entity in a property definition in the same `MERGE` clause is deprecated.
```

**Comment by neo-technology-commit-status-publisher** [comment]
Thanks for the documentation updates.

The preview documentation has now been torn down - reopening this PR will republish it.

**Pull request merged**